server-common/src/main/java/org/apache/kafka/server/config/ConfigType.java

```diff
@@ -31,6 +31,5 @@ public class ConfigType {
     public static final String CLIENT_METRICS = "client-metrics";
     public static final String GROUP = "groups";
 
-    // Do not include ClientMetrics and Groups in `all` as they are not supported on ZK.
```

Comment:
Hmm, does this mean we should change `ALL`? cc @apoorvmittal10 @AndrewJSchofield

Reply:
Perhaps we can eliminate `ALL` since it's exclusively used by `ConfigCommand` after the removal of ZooKeeper code

Reply:
Why would we remove it? It seems like a useful concept and better placed here than in the command itself.

Reply:
I would, in fact, avoid the need for `ConfigCommand` to do its own definition of the config types the system supports - commands should ideally not have any such logic.

Reply:
`ConfigCommand` use `ALL` to do pre-check before calling related APIs. Hence, it should be fine to remove the `ALL` as  `ConfigCommand` eventually throw exception if user-defined string is not a allowed type.

Reply:
Right - it's a very common pattern that when you have an enumeration, you provide a mechanism to know all of the values in that enumeration. I don't know why we're debating that here.

Reply:
Yes, I suggest changing the definition of `ALL` here and then updating https://github.com/apache/kafka/blob/6eb44ad86994b92edd3c0d807830b79807adda59/core/src/main/scala/kafka/admin/ConfigCommand.scala#L66

Reply:
If we want to keep it, could we consider using enum instead? It can ensure `ALL` include all items as we can leverage enum function to implement it.

Reply:
+1 to replace with current implementation with enum.

Reply:
If we agree to convert this class to an enum, I suggest we handle this after #18585 is merged, since `ConfigType` is widely used by `AdminZkClient`.

Reply:
Making it an enum is fine, but can also be done separately. We can file that as a JIRA if we want to decouple from this PR. The reason why I raised the issue is that by removing the comment that was there, we had lost track of the reason for the unusual behavior (which is no longer needed).

Reply:
I see. 
Indeed the definition of All has been change after removing zk.
I will file a ticket for converting to enum issue.

Reply:
https://issues.apache.org/jira/browse/KAFKA-18576